### PR TITLE
[FIX] Unable to subscribe somebody to a template

### PR DESCRIPTION
**Pull request by ajayjaviya**
Task: https://www.druidoo.io/web#id=2152&action=361&active_id=28&model=project.task&view_type=form&menu_id=237

### CODE REVIEW

File: coop_shift/model/shift_template_registration_line.py [hunk 1]
```diff
@@ -65,8 +65,8 @@ def _check_over_lap(self):
         self.ensure_one()
         lines = self.partner_id.tmpl_reg_line_ids
         for line in lines:
-            if (line.date_begin <= self.date_end or not self.date_end) and\
-                (line.date_end >= self.date_begin or not line.date_end) and\
+            if (not self.date_end or line.date_begin <= self.date_end) and\
+                (not line.date_end or line.date_end >= self.date_begin) and\
                     line.id != self.id:
```
Comment (ivantodorovich) [inline coop_shift/model/shift_template_registration_line.py]:
```suggestion
            if (
                line.id != self.id
                and (not self.date_end or line.date_begin <= self.date_end)
                and (not line.date_end or line.date_end >= self.date_begin)
            ):
```

**Comment by ivantodorovich** [comment]
Can you also add a test case to add coverage in `_check_over_lap` ?

**Comment by ivantodorovich** [comment]
@ajayjaviya can you please rebase on top of 12.0?
This branch has merging conflicts.

Also there's something weird, I see many commits in this PR that do not belong here.
I think you probably rebased and pushed without --force ?

**Comment by ajayjaviya** [comment]
> @ajayjaviya can you please rebase on top of 12.0?
> This branch has merging conflicts.
> 
> Also there's something weird, I see many commits in this PR that do not belong here.
> I think you probably rebased and pushed without --force ?

@ivantodorovich : I have rebase the branch with 12.0

**Pull request merged**